flink-core/src/main/java/org/apache/flink/core/fs/local/LocalFileSystem.java

```diff
@@ -262,8 +265,13 @@ public FSDataOutputStream create(
 	public boolean rename(final Path src, final Path dst) throws IOException {
 		final File srcFile = pathToFile(src);
 		final File dstFile = pathToFile(dst);
-
-		return srcFile.renameTo(dstFile);
+		//Files.move fails if the destination directory doesn't exist
+		if(dstFile.getParentFile()!= null && !dstFile.getParentFile().exists()){
+			Files.createDirectories(dstFile.getParentFile().toPath());
+		}
+		//this throws an IOException if any error occurs
+		Files.move(srcFile.toPath(), dstFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
If we want to preserve the old semantics, we should catch the exception and return `false` in that case.